state/options/tx_options.go

```diff
@@ -0,0 +1,201 @@
+package options
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/celestiaorg/celestia-app/pkg/user"
+	apptypes "github.com/celestiaorg/celestia-app/x/blob/types"
+)
+
+const (
+	// gasMultiplier is used to increase gas limit in case if tx has additional options.
+	gasMultiplier = 1.1
+
+	//	Since 0 is a valid fee input for the Tx, the default value is -1.
+	defaultFeeAmount = -1
+)
+
+var (
+	errNoGasProvided     = errors.New("gas limit was not set")
+	errNoAddressProvided = errors.New("address was not set")
+)
+
+// TxOptions specifies additional options that will be applied to the Tx.
+type TxOptions struct {
+	// fee is private since it has to be set through `SetFeeAmount`
+	fee      *Fee
+	GasLimit uint64
+
+	// Specifies the key from the keystore associated with an account that
+	// will be used to sign transactions.
+	// NOTE: This `Account` must be available in the `Keystore`.
+	Account string
+	// Specifies the account that will pay for the transaction.
+	// Input format Bech32.
+	Granter string
+}
```

Comment:
I think we can make these fields a little more verbose and clear for the user. I would propose:
- renaming `Granter` to `FeePayingAddress`
- renaming `Account` to `SigningAccount`

I've tried to only use the term `Address` when referring to the Bech32 encoded SDK address and `Account` to refer to the key name that links to the underlying `Record` with the users private & public keys.

Reply:
> Account to refer to the key name that links to the underlying Record with the users private & public keys.
The account refers to the key name in the key store.

> renaming Granter to FeePayingAddress
Account will also pay fees iirc, so it may be confusing for the user.

Reply:
I would also prefer keeping it as Account + Granter

But maybe Account + GranterAddress would be clearer. But no strong opinions here

Reply:
> The account refers to the key name in the key store.

Yes that's exactly what I was saying (`Record` comes from the Keyring)

> Account will also pay fees iirc, so it may be confusing for the user.

Then maybe be more specific in the comments

> I would also prefer keeping it as Account + Granter

I don't think `Granter` by itself is sufficient. Imagine users who have no prior context. At least `FeeGranter`

Reply:
FeeGranterAddress sounds good to me.